### ISSUE TITLE
Missing Exception Handling in pymssql for MSSQL Server Stored Procedure Errors

### ISSUE THREAD

**Issue by DmitryGrb** [pymssql/pymssql#877]
The MSSQL server supports multiple result sets as output of a stored procedure. However, if an exception occurs on the SQL side after the first result set is generated, no exceptions are generated on the client side in the pymssql code.

**To Reproduce**
Create the following stored procedure on MSSQL server:
```
create procedure [dbo].[sp1]
as
    select 1
    select 1/0
;
GO
```
Execute the stored procedure via Python:
```
import pymssql

conn = pymssql.connect(host, user, password, db)
cursor = conn.cursor()
cursor.execute('exec [dbo].[sp1]')
```

**Expected behavior**

An error should be thrown with the following message:

```Divide by zero error encountered.DB-Lib error message 20018, severity 16: General SQL Server error: Check messages from the SQL Server```

**Current Behavior**

The program exits with a 0 exit code and no warning:
```Process finished with exit code 0```

**Context (Environment)**

**Originally observed in:**

- AWS Glue Python 3.9 shell job (AWS Glue 4.0)
- pymssql 2.2.7

**Reproduced on**

System:
platform.system()       : Darwin
platform.architecture() : ('64bit', '')
platform.libc_ver()     : ('', '')
Python:
sys.version             : 3.9.6 (default, Oct 18 2022, 12:41:40) 
[Clang 14.0.0 (clang-1400.0.29.202)]
pymssql:
VERSION            : (2, 2, 7)
version            : 2.2.7
full_version       : 2.2.7
dbversion          : freetds v1.4.6
_mssql:
VERSION            : (2, 2, 7)
version            : 2.2.7
full_version       : 2.2.7
dbversion          : freetds v1.4.6
login_timeout      : 60
min_error_severity : 6

**Comment by gordthompson** [comment]
The behaviour you are seeing is a result of how the TDS (Tabular Data Stream) protocol works. The first result set (from `SELECT 1`) is automatically made available for us to process, e.g., by using `cursor.fetchall()`

```
/home/gord/PycharmProjects/pymssql_demo/venv/bin/python /home/gord/PycharmProjects/pymssql_demo/main.py 
> /home/gord/PycharmProjects/pymssql_demo/main.py(12)<module>()
-> rows = cursor.fetchall()
(Pdb) next
> /home/gord/PycharmProjects/pymssql_demo/main.py(13)<module>()
-> cursor.nextset()
(Pdb) 
```

We can verify the rows returned by inspecting our `rows` variable

```
(Pdb) rows
[(1,)]
(Pdb) 
```

The second result set is queued until we call `cursor.nextset()`. If that result contains an error code then a pymssql exception is thrown:

```
(Pdb) l 13,13
 13  ->	cursor.nextset()
(Pdb) next
pymssql._pymssql.OperationalError: (8134, b'Divide by zero error encountered.DB-Lib error message 20018, severity 16:\nGeneral SQL Server error: Check messages from the SQL Server\n')
> /home/gord/PycharmProjects/pymssql_demo/main.py(13)<module>()
-> cursor.nextset()
(Pdb)
```

So, if your stored procedure is going to return multiple result sets you need to make sure that you call `.nextset()` enough times to process them all.

**Comment by termim** [comment]
@gordthompson  Excellent explanation, would you be able to  do a PR for the docs (maybe in FAQ)?

**Comment by termim** [comment]
Closed in #879
